### ISSUE TITLE
Proposal to redo frontend

### ISSUE THREAD

**Issue by ReubenMenezes** [nhcommunity/homepage#42]
### What feature would you like to see?

Upon reviewing the frontend of the site, I noticed that we've cramped a lot of stuff together. There are currently a few problems.
- The site is rather unresponsive 
- We're using stuff like the HTML `<b>` tag which isn't supposed to be used anymore ( it's HTML 4 and while some validators find it to be fine, it's an issue for accessibility
- We're storing all the data of the mods in `.json` files. This does work. However it would be way easier to make an object called users and have the data filled in there. 
- We also seem to be having a bit of a mess with the ids and classes atm. I suggest we adopt the [BEM methodology](http://getbem.com/introduction/) for our CSS
- We also aren't using semantic elements. This isn't the biggest issue but it's practically far better to have 
```html
<nav class ="nav">
```
instead of 
```html
<ul class ="nav">
```
Makes it easier for a11y and also makes stuff a bit easier.
All of this stuff can be fixed. However instead of going off the current version of our page and putting out many fires (which may take some time) I suggest we create a new branch for the rewrite. Some things that I hope to implement are
- resources.html page
- a bit of a cleaner folder structure, the project has a rather unintuitive folder structure at the moment
- Using the github api to update the mods profile photos (at least for those who have kept their photos as their github photos in the mods section)
- Completing the mods section, it's incomplete at the moment

As per Nick's wishes, the project shall still use HTML,CSS and JS instead of preprocessors like SASS etc.
This is because he wishes to have the project as simple as possible for folks who are just starting their journey

Thoughts everyone?

### Additional information

I do have a few ideas using intersection observer but those are definitely off the table right now, there are more important things.

**Comment by nhcarrigan** [comment]
This all sounds good. I've created the `rewrite` branch for folks to work off of.

> Using the github api to update the mods profile photos (at least for those who have kept their photos as their github photos in the mods section)

This can be done without an API call through the `.png` trick: `https://github.com/nhcarrigan.png`.

**Comment by ReubenMenezes** [comment]
> This can be done without an API call through the .png trick: https://github.com/nhcarrigan.png.

I see, sounds good.

**Comment by ReubenMenezes** [comment]
So, are we assigning this issue or are we leaving it open to the community? I've already started working on it

**Comment by nhcarrigan** [comment]
Yeah this isn't necessarily something that can be fixed in a single pull request, I think.

Our community is pretty quiet, though - if you're interested in spearheading this work I can definitely assign the issue to you.

**Comment by ReubenMenezes** [comment]
I'd love to

**Comment by nhcarrigan** [comment]
Assigned!

**Comment by Asmit2952** [comment]
Can I also work on this issue?

**Comment by ReubenMenezes** [comment]
We just merged a pr recently, theres a layout issue. You can open up another issue about the layout and work on that

**Comment by AnujSsStw** [comment]
hey, I would to love collab. if still room for me

**Comment by ReubenMenezes** [comment]
Hey there, of course there is! Just read the messages in this pr about the layout issue. Then just create another issue for it and ask to be assigned to it. The top messages of this thread include the goals. Good Luck!

**Comment by TAKANOME-DEV** [comment]
Hey folks, still rewriting the frontend ?? 
How can I help ??

**Comment by ReubenMenezes** [comment]
Indeed! There's currently a bit of an issue with the layout (I think I screwed something up while handling a merge conflict). Check out #46. I've mentioned you there

**Comment by TAKANOME-DEV** [comment]
Cool, gonna check that.

**Comment by TAKANOME-DEV** [comment]
Hi there, so i fixed the ui issue in the rewrite branch. See #54 
I was wondering if we can add the social links for the team inside the card to make it more easy instead of the modcard??

**Comment by shubhamsigdar1** [comment]
Hey, I want to contribute to this project

**Comment by ReubenMenezes** [comment]
We're done with contributions to this issue, sorry.

**Comment by shubhamsigdar1** [comment]
ok...I see that the nav bar or header is not responsive

**Comment by shubhamsigdar1** [comment]
You can create any issue on that

**Comment by ReubenMenezes** [comment]
Feel free to create an issue.
